voluncheer/map/management/commands/pull_nyc_charities.py

```diff
@@ -0,0 +1,39 @@
+from django.core.management.base import BaseCommand
+from django.core.management.base import CommandError
+from django.db import transaction
+import requests
+
+from map.models import NYCharities
+
+
+class Command(BaseCommand):
+    help = "Pull data from NYC Open Data and save to NYCharities table"
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        url = "https://data.cityofnewyork.us/resource/yunp-vs8g.json"
+
+        response = requests.get(url)
+
+        if response.status_code != 200:
+            raise CommandError("Could not pull data from NYC Open Data")
+
+        # Clear NYCharities table before inserting new data
+        NYCharities.objects.all().delete()
```

Comment:
Deleting all of the existing entries may not be supportable when using more durable storage. Since it will essentially change the primary key out from under us, if that key is referenced anywhere else that link will break. We should support updating existing entries and creating new ones. As far as removing old ones we can probably ignore that for now.

WDYT?

Reply:
Makes sense! I will change it to update existing entries and create new ones.

Reply:
Great, let me know when that's done and I can re-review.

Reply:
Done! Now it doesn't delete the objects, but it uses the update_or_create method to find a charity with the provided info, and updates the defaults if found. I had to add a bin_num attribute because it was unique for each entry in the data set. This way, if a charity changes its location, we can update the new address and show this on the map.